src/DynamicExpresso.Core/Parsing/ParserSettings.cs

```diff
@@ -33,6 +33,19 @@ public ParserSettings(bool caseInsensitive,bool lateBindObject)
 			LambdaExpressions = false;
 		}
 
+		/// <summary>
+		/// Creates a deep copy of the current settings, so that the identifiers/types/methods can be changed
+		/// without impacting the existing settings.
+		/// </summary>
+		internal ParserSettings Clone()
```

Comment:
I usually implement clone method using the "copy constructor" pattern.
I think it is easier and we can leave readonly on the variables.

Essentially we can have an additional constructor like:

```
private ParserSettings(ParserSettings other)
{
// TODO Here we can copy the variables
}
```

And then the clone method can be implemented as:

```
internal ParserSettings Clone()
{
return new ParserSettings(this);
}
```

We can also eval to put `Clone` as public. What do you think?

Reply:
The reason I went without a copy constructor is to be able to use the `MemberwiseClone()` method, which creates a shallow copy of all fields. Then, I can new instances of the various lists when needed. My main fear with a copy constructor is always to forget to update it when a new field is added..

I don't have a strong opinion either way, so I'll implement your suggestion and make it public :)

Reply:
Yes, I get your point of view. My fealing is that removing `readonly` can introduce more strange bugs. But also for me this is not a strong opinion. As you prefer.

Reply:
I agree with your assessment that removing the `readonly` modifier can lead to strange bugs, so I added the copy ctor.